src/wallet/wallet2.cpp

```diff
@@ -4329,7 +4329,17 @@ bool wallet2::load_keys_buf(const std::string& keys_buf, const epee::wipeable_st
   {
     if (encrypted_secret_keys)
     {
-      m_account.decrypt_keys(key);
+      m_account.decrypt_legacy(key); // try legacy decryption
```

Comment:
I haven't looked closely enough at the new format to be certain if this would work, but, if possible, you may want to switch the ordering here to assume the new format and fall back to the old one, that way everyone upgrades once and then stops wasting cycles on the old format, rather than doing wasted work in perpetuity.

Reply:
This order is swapped in a pending commit.